hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/webapp/dao/NodeToLabelsEntry.java

```diff
@@ -44,6 +44,11 @@ public NodeToLabelsEntry(String nodeId, ArrayList<String> labels) {
     this.labels = labels;
   }
 
+  public NodeToLabelsEntry(String nodeId, Collection<String> labels) {
+    this.nodeId = nodeId;
+    this.labels.addAll(labels);
```

Comment:
Do we need to clean the existing?
This usually also complains as name clashing, we should probably use pLabels for the arg.

Reply:
Thank you very much for helping to review the code, I will modify the code.

Reply:
This function is the constructor, we don't need to clean up the existing data.